lib/runner/extract-runnable-items.js

```diff
@@ -186,6 +187,19 @@ var sdk = require('postman-collection'),
 
         items = findItemsOrGroups(collection, entrypointLookup, true);
 
+        // Sorting the items and folders by the order in which they appear as folder/request arguments,
+        // only if specified in the entrypoint.preserveOrder
+        preserveOrder && items && items.members && Array.isArray(items.members) && items.members.sort(function (a, b) {
```

Comment:
Instead of `sort`, why don't we just traverse the `entrypoints` array and get the reference `Item`s from `items.reference`?

Reply:
@codenirvana 
Sure, I agree with this actually.